### PR TITLE
Add GlobalAvgPool1D Layer

### PR DESCRIPTION
**Pull request by therealansh**
This PR adds GlobalAvgPool1D layer (#63). Also renamed the file to "GlobalAvgPooling2D" from "GlovalAvgPooling2d"(typo).

TODO:
- [x] Add layer implementation
- [x] Add support for export and import of layer in JSON format 
- [x] Add Documentation
- [x] Add Tests

### CODE REVIEW

**Comment by therealansh** [comment]
The unit tests written right now are only for GlobalAvgPool1D extending the PoolingTest class. With the help of helper functions written by @avan1235 in the ShapeFunctions ``` PoolLayerTest``` can be used as an open class for testing more Pooling layers in future.

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * Input shape: 3D tensor with shape `(batch_size, steps, features)`.
+ *
+ * Output shape: 2D tensor with shape `(batch_size, features)`.
+ *
+ * @property [name] Custom layer name.
+ * @constructor Creates [GlobalAvgPool1D] object.
+ *
+ */
+
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt:28]: Please, remove the blank line

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * NOTE: Works with tensors which must have rank 3 (batch, steps, features).
+ *
+ * Input shape: 3D tensor with shape `(batch_size, steps, features)`.
+ *
+ * Output shape: 2D tensor with shape `(batch_size, features)`.
+ *
+ * @property [name] Custom layer name.
+ * @constructor Creates [GlobalAvgPool1D] object.
+ *
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt:26]: Please remove the blank line

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        name: String = ""
+) : Layer(name) {
+    override fun build(tf: Ops, kGraph: KGraph, inputShape: Shape) { }
+
+    override fun computeOutputShape(inputShape: Shape): Shape {
+        return Shape.make(inputShape.size(0), inputShape.size(2))
+    }
+
+    override fun forward(tf: Ops, input: Operand<Float>, isTraining: Operand<Boolean>, numberOfLosses: Operand<Float>?): Operand<Float> {
+        var stepAxis = 1                //if dataFormat == "channels_last", it will be 2 if "channels_first"
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalAvgPool1D.kt:39]: please mark it with the TODO: for the future (when both formats will be supported)

File: api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PoolLayerTest.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    protected fun assertGlobalAvgPool1DEquals(
+        layer: Layer,
+        input:Array<Array<FloatArray>>,
+        expected: Array<FloatArray>,
+    ) {
+        val actual = Array(expected.size) {FloatArray(expected[0].size) { 0.toFloat() } }
+        assertPoolingLayer(layer,input, expected,actual,::assertGlobalAvgPool1DEquals)
+    }
+
+
```
Comment (zaleslaw) [inline api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PoolLayerTest.kt:25]: please remove blank line

File: api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PoolLayerTest.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import org.junit.jupiter.api.Assertions.assertArrayEquals
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.tensorflow.EagerSession
+import org.tensorflow.Graph
+import org.tensorflow.*
+import org.tensorflow.op.Ops
+
+
+open class PoolLayerTest {
+
```
Comment (zaleslaw) [inline api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PoolLayerTest.kt:15]: please remove blank line

File: api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalAvgPooling1DTest.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package org.jetbrains.kotlinx.dl.api.core.layer
+
+import org.jetbrains.kotlinx.dl.api.core.layer.pooling.GlobalAvgPool1D
+import org.junit.jupiter.api.Test
+
+internal class GlobalAvgPooling1DTest : PoolLayerTest() {
+
```
Comment (zaleslaw) [inline api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalAvgPooling1DTest.kt:7]: please remove blank line

File: api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PoolLayerTest.kt [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                expected.size.toLong(),
+                expected[0].size.toLong()
+            )
+
+            val actualShape = shapeFromDims(*output.shape())
+            output.copyTo(actual)
+            assertEquals(expectedShape, actualShape)
+            assertEqual(expected,actual)
+        }
+
```
Comment (zaleslaw) [inline api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PoolLayerTest.kt:53]: please remove blank line

**Review by zaleslaw — APPROVED** [review]
So, this LGTM for me also I'd like the tests, it's a great implementation and very clear for now
I suggest removing some blank lines in some places, but it's ready for merge.

There is a queue of PRs, I'll notify you when you should merge master into your branch (in a few days) and your PR will be merged

**Comment by zaleslaw** [comment]
So you could merge the master branch and I'll merge your PR in two days

**Comment by zaleslaw** [comment]
Congrats, @therealansh thanks for the contribution!

**Pull request merged**